packages/web-components/fast-components/src/switch/switch.styles.ts

```diff
@@ -57,214 +51,159 @@ export const switchStyles: (
              * Chromium likes to select label text or the default slot when
              * the checkbox is clicked. Maybe there is a better solution here?
              */ ""
-        } user-select: none;
+        }
+        user-select: none;
+        width: max-content;
     }
-
-    :host([disabled]) {
+    :host(.disabled) {
         opacity: ${disabledOpacity};
     }
-
-    :host([disabled]) .label,
-    :host([readonly]) .label,
-    :host([readonly]) .switch,
-    :host([disabled]) .switch {
+    :host(.disabled) .label,
+    :host(.readonly) .label,
+    :host(.disabled) .switch,
+    :host(.readonly) .switch,
+    :host(.disabled) .status-message,
+    :host(.readonly) .status-message {
         cursor: ${disabledCursor};
     }
-
     .switch {
         position: relative;
         outline: none;
         box-sizing: border-box;
-        width: calc(${heightNumber} * 1px);
-        height: calc((${heightNumber} / 2 + ${designUnit}) * 1px);
+        width: calc((${heightNumber} - (${designUnit} * 2 )) * 1px);
+        height: calc((${heightNumber} / 2) * 1px);
         background: ${neutralFillInputRest};
-        border-radius: calc(${controlCornerRadius} * 1px);
-        border: calc(${strokeWidth} * 1px) solid ${neutralStrokeRest};
+        border-radius: calc(${heightNumber} * 1px);
+        border: calc(${strokeWidth} * 1px) solid ${strokeControlStrongRest};
     }
-
-    .switch:hover {
-        background: ${neutralFillInputHover};
-        border-color: ${neutralStrokeHover};
-        cursor: pointer;
+    :host([aria-checked="true"]) .switch {
+        border-color: ${strokeControlStrongRest};
+        background: ${neutralFillInputActive};
     }
-
-    host([disabled]) .switch:hover,
-    host([readonly]) .switch:hover {
-        background: ${neutralFillInputHover};
-        border-color: ${neutralStrokeHover};
-        cursor: ${disabledCursor};
+    :host(:enabled:hover) .switch,
+    :host([aria-checked="true"]:enabled:hover) .switch {
+        border-color: ${strokeControlStrongHover};
     }
-
-    :host(:not([disabled])) .switch:active {
+    :host(:enabled:active) .switch,
+    :host([aria-checked="true"]:enabled:active) .switch {
         background: ${neutralFillInputActive};
-        border-color: ${neutralStrokeActive};
+        border-color: ${strokeControlStrongActive};
     }
-
-    :host(:${focusVisible}) .switch {
-        box-shadow: 0 0 0 2px ${fillColor}, 0 0 0 4px ${focusStrokeOuter};
+    :host(:${focusVisible}) .switch,
+    :host([aria-checked="true"]:${focusVisible}:enabled) .switch {
+        border-color: ${strokeControlStrongFocus};
+        box-shadow: 0 0 0 calc(${focusStrokeWidth} * 1px) ${strokeControlStrongFocus};
     }
-
-    .checked-indicator {
+    slot[name="switch"] {
```

Comment:
Did you test that this works when you _slot_ an item as well? The way this is written it looks like this would slot the default content, but I'm not sure it would catch something slotted.

Reply:
I tested adding a slotted SVG
```html
<fast-switch>
        Custom switch indicator
        <svg
            slot="switch"
            width="8"
            height="8"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
        >
            <path d="M5.28347 1.54556C5.57692 0.95096 6.42479 0.950961 6.71825 1.54556L7.82997 3.79817L10.3159 4.15939C10.9721 4.25474 11.2341 5.06112 10.7592 5.52394L8.96043 7.27736L9.38507 9.75321C9.49716 10.4067 8.81122 10.9051 8.22431 10.5966L6.00086 9.42761L3.7774 10.5966C3.19049 10.9051 2.50455 10.4067 2.61664 9.75321L3.04128 7.27736L1.24246 5.52394C0.767651 5.06111 1.02966 4.25474 1.68584 4.15939L4.17174 3.79817L5.28347 1.54556Z"/>
    </svg>
</fast-switch>
```
![image](https://user-images.githubusercontent.com/37851220/142508134-b47e1a9f-6cd7-4082-b3a3-9fa44eaeab63.png)
![image](https://user-images.githubusercontent.com/37851220/142508360-abbf7f2c-2060-40b5-81eb-8038f35959c3.png)